tools/python/boutiques/localExec.py

```diff
@@ -279,40 +279,50 @@ def execute(self, mount_strings):
                 envString = " "
                 if envVars:
                     for (key, val) in list(envVars.items()):
                         envString += " -e {0}='{1}' ".format(key, val)
+
                 # export mounts to docker string
+                # Warning: mount_strings should have at least one entry ()
+                # otherwise docker run will failed
                 docker_mounts = " -v ".join(m for m in mount_strings)
+
                 # If --changeUser was desired, provides the current user id
                 # and its group id as the user and group to be used instead
                 # of the default root within the container.
                 userchange = ''
                 if self.changeUser:
                     userchange = ' -u $(id -u):$(id -g)'
 
-                container_command = ('docker run' + userchange +
-                                     ' --entrypoint=' + self.shell +
-                                     ' --rm' + envString +
-                                     ' -v ' + docker_mounts +
-                                     ' -w ' + launchDir + ' ' +
-                                     conOptsString +
-                                     str(conImage) + ' ' + dsname)
+                container_command = ('docker run'     + userchange    +
+                                     ' --entrypoint=' + self.shell    +
+                                     ' --rm'          + envString     +
+                                     ' -v '           + docker_mounts +
+                                     ' -w '           + launchDir     + ' ' +
+                                     conOptsString    +
+                                     str(conImage)    + ' '           + dsname
+                                    )
+
             elif conTypeToUse == 'singularity':
                 envString = ""
                 if envVars:
                     for (key, val) in list(envVars.items()):
                         envString += "SINGULARITYENV_{0}='{1}' ".format(key,
                                                                         val)
-                singularity_mounts = '-B ' + ' -B '.join(mount_strings)
-                container_command = (envString + 'singularity exec '
-                                     '--cleanenv ' +
-                                     singularity_mounts +
-                                     ' -W ' + launchDir + ' ' +
-                                     conOptsString +
-                                     str(conPath) + ' ' + dsname)
+                # Warning: mount_strings should have at least one entry ()
+                # otherwise singularity run will failed
+                singularity_mounts = ' -B '.join(mount_strings)
+                container_command  = (envString               + 'singularity exec ' +
+                                     '--cleanenv '            +
+                                     ' -B '                   + singularity_mounts  +
```

Comment:
I'm not sure how that fixes anything, you removed "-B" from l318 and put it back here.

Reply:
It was tested... But just realize that your dev branch has the fix for the issue #540.